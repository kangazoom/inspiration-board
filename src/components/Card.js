import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);
  };


  render() {

    console.log(this.props.card.id)
    let id = this.props.card.id;
    console.log(id)
    let text = this.props.card.text;
    let emoji = this.props.card.emoji;


    return (
      <div className="card" key={id}>
        <p >Text: "{text}"</p>
        <p>Emoji: {emoji}</p>
      </div>
    )
  }
}
//
// Card.propTypes = {
//
// };

export default Card;
