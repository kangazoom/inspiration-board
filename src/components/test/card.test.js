import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';


describe('card', () => {
  it('will match the Card Snapshot', () => {
    const cardTester = {
      id: 1,
      text: 'testing is fun!'
    }
    const wrapper = shallow(
      <Card card={cardTester}/>
    );

    expect(wrapper).toMatchSnapshot();
  })
});
