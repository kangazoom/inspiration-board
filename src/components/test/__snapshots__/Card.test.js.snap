// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the Card Snapshot 1`] = `
<div
  className="card"
  key="1"
>
  <p>
    Text: "
    testing is fun!
    "
  </p>
  <p>
    Emoji: 
  </p>
</div>
`;
